workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.ts

```diff
@@ -0,0 +1,131 @@
+   *
+   * @returns - {@link CreateGroupResponse}
+   *
+   * @throws - {@link GroupAlreadyExistsError} Can not create a group that already exists
+   */
+  public async createGroup(
+    user: AuthenticatedUser,
+    createGroupRequest: CreateGroupRequest
+  ): Promise<CreateGroupResponse> {
+    throw new Error('NotImplemented');
+  }
+  /**
+   * Delete an authorization group
+   * @param user - {@link AuthenticatedUser}
+   * @param deleteGroupRequest - {@link DeleteGroupRequest}
+   *
+   * @returns - {@link DeleteGroupResponse}
+   */
+  public async deleteGroup(
+    user: AuthenticatedUser,
+    deleteGroupRequest: DeleteGroupRequest
+  ): Promise<DeleteGroupResponse> {
+    throw new Error('NotImplemented');
+  }
+  /**
+   * Create identity permissions
+   * @param user - {@link AuthenticatedUser}
+   * @param createIdentityPermissionsRequest - {@link CreateIdentityPermissionsRequest}
+   *
+   * @returns - {@link CreateIdentityPermissionsResponse}
+   *
+   * @throws - {@link IdentityPermissionAlreadyExistsError} Can not create an identity permission that already exists.
+   */
+  public async createIdentityPermissions(
+    user: AuthenticatedUser,
+    createIdentityPermissionsRequest: CreateIdentityPermissionsRequest
+  ): Promise<CreateIdentityPermissionsResponse> {
+    throw new Error('NotImplemented');
+  }
+  /**
+   * Delete identity permissions
+   * @param user - {@link AuthenticatedUser}
+   * @param deleteIdentityPermissionsRequest - {@link DeleteIdentityPermissionsRequest}
+   *
+   * @returns - {@link DeleteIdentityPermissionsResponse}
+   */
+  public async deleteIdentityPermissions(
+    user: AuthenticatedUser,
+    deleteIdentityPermissionsRequest: DeleteIdentityPermissionsRequest
+  ): Promise<DeleteIdentityPermissionsResponse> {
+    throw new Error('NotImplemented');
+  }
+  /**
+   * Delete all permissions associated to the subject
+   * @param user - {@link AuthenticatedUser}
+   * @param deleteSubjectPermissionsRequest - {@link DeleteSubjectPermissionsRequest}
+   */
+  public async deleteSubjectPermissions(
+    user: AuthenticatedUser,
+    deleteSubjectPermissionsRequest: DeleteSubjectPermissionsRequest
+  ): Promise<DeleteSubjectPermissionsResponse> {
+    throw new Error('NotImplemented');
+  }
+  /**
+   * Assign a user to an authorization group
+   * @param user - {@link AuthenticatedUser}
+   * @param assignUserToGroupRequest - {@link AssignUserToGroupRequest}
+   */
+  public async assignUserToGroup(
+    user: AuthenticatedUser,
+    assignUserToGroupRequest: AssignUserToGroupRequest
+  ): Promise<AssignUserToGroupResponse> {
+    throw new Error('NotImplemented');
+  }
+  /**
+   * Remove a user from an authorization group
+   * @param user - {@link AuthenticatedUser}
+   * @param removeUserFromGroupRequest - {@link RemoveUserFromGroupRequest}
+   */
+  public async removeUserFromGroup(
+    user: AuthenticatedUser,
+    removeUserFromGroupRequest: RemoveUserFromGroupRequest
```

Comment:
for all of these, should `user` be included in the `Request` object? That way its the only parameter?

Reply:
The `Request` object is an input for the dynamicPermissionsPlugin, I didn't find it necessary to extract it into a single request object